src/ConfirmDeleteAllModal.js

```diff
@@ -18,43 +18,31 @@ type ConfirmDeleteAllModalProps = {
 class ConfirmDeleteAllModal extends React.Component<ConfirmDeleteAllModalProps, {}> {
     render() {
         return (
-            <Modal
-                aria-labelledby='deleteAll-button'
+            <ModalWithFooter
                 show={this.props.show}
-                onHide={this.props.onCancel}
-                size='lg'
-                dialogClassName='ConfirmDeleteAllModal'
-                centered>
-                <Modal.Body className='ConfirmDeleteAllModal__content'>
-                    <div className='ConfirmDeleteAllModal__header'>
-                        <span role='img' aria-hidden='true' >
-                            <ErrorIcon className='ConfirmDeleteAllModal__warning-svg' />
-                        </span>
-                        <FormattedMessage id='ConfirmDeleteAllModal.title' />
-                    </div>
-                    <div className='ConfirmDeleteAllModal__footer'>
-                        <Button
-                            className='ConfirmDeleteAllModal__option-button cancel'
-                            onClick={this.props.onCancel}>
-                            <FormattedMessage id='ConfirmDeleteAllModal.cancelButton' />
-                        </Button>
-                        <Button
-                            id='deleteAll-button'
-                            className='ConfirmDeleteAllModal__option-button confirm'
-                            onClick={this.props.onConfirm}>
-                            <FormattedMessage id='ConfirmDeleteAllModal.confirmButton' />
-                        </Button>
-                    </div>
-                </Modal.Body>
-            </Modal>
+                focusElementSelector={'#DeleteAll-done'}
+                focusOnCloseSelector={'.ProgramBlockEditor__program-deleteAll-button'}
+                ariaLabel={this.props.intl.formatMessage({id: 'ConfirmDeleteAllModal.confirmButton'})}
+                onClose={this.props.onCancel}
+                buttonProperties={
+                    [
+                        {label: this.props.intl.formatMessage({id: 'ConfirmDeleteAllModal.cancelButton'}), onClick: this.props.onCancel, isPrimary: false},
```

Comment:
I still think we could find a way to not have to indicate that `isPrimary` is `false`.

Reply:
I think it is safe to assign false this way, because we are checking isPrimary property to decide className for a specific button, and if we make isPrimary optional, the logic for the decision would still work, but I think using negation on undefined little unsafe.